bash_completion

```diff
@@ -1480,13 +1528,18 @@ _available_interfaces()
 }
 
 # Echo number of CPUs, falling back to 1 on failure.
-# TODO:API: rename per conventions, rework to use vars rather than outputting
-_ncpus()
+# @var[out] ret
+_comp_get_ncpus()
 {
     local var=NPROCESSORS_ONLN
     [[ $OSTYPE == *@(linux|msys|cygwin)* ]] && var=_$var
-    local n=$(getconf $var 2>/dev/null)
-    printf %s "${n:-1}"
+    ret=$(getconf $var 2>/dev/null)
+
+    # make sure that it only contains number so that it can be safely used in
+    # comgpen -W "...".
+    ret=${ret//[^0-9]/}
+
+    ret=${ret:-1}
```

Comment:
I like the idea. Also, I think it produces unexpected results when the function returns `0`. The brace expansion still produces results for reverse ranges like `{1..0}`.

Reply:
Maybe we can fall back to `ret=1` on failure because we know that we have at least one core as far as Bash is running.

Reply:
I still fall back to `1`. In addition, we set 1 when the result is less than 1.

I also added a fix for `cppcheck` fd1c98b2bce4d071f30395ad3ae9d60c1100c9c3, which produced 2 and 1 by the reverse range of the brace expansions.

Reply:
I wonder why producing from `{2..1}` (in descending order) would be a problem? They end up sorted when the completions are displayed anyway.

On the other hand, I'm not sure how much of a use case it is to include `1` among the completions, but I'm leaning towards including it being better than offering no completions at all for `-j` unless more than 2 CPUs are found.

(The original implementation of mine hardcoded `{2..16}` and that was later changed to `{2..$(_ncpus)}`, also by me.)

Anyway I think I'd go with the usual `_comp_get_ncpus && _comp_compgen -- -W "{1..$ret}"` to remove the weirdness.

Reply:
I don't mind the ordering. The problem is that `2` is always generated even if there is only one core. Also, the intent appeared to be to generate the numbers >= 2, but `{2..1}` generates also 1.

> Anyway I think I'd go with the usual `_comp_get_ncpus && _comp_compgen -- -W "{1..$ret}"` to remove the weirdness.

I changed it to include 1 as other cases do ([`e7231c8..d5a78ce`](https://github.com/scop/bash-completion/compare/e7231c8cd2ae47a4af0c92724c1c0b71922dbfb2..d5a78ce0567686e34764ad4b804430d4f917e782#diff-86eb7176fdea02ffa0269186fb0908948c334477000ab7a3754ef9a4bf0f8eff)).

> (The original implementation of mine hardcoded `{2..16}` and that was later changed to `{2..$(_ncpus)}`, also by me.)

I'm now generating 1 even when `_comp_get_ncpus` failed because we can assume that there is at least one core. But as was done before introducing `_ncpus`, we might generate a larger number as the default value on failure.

<!-- Hmm, but maybe it is better to generate no numbers at all so that the user can decide the number. Inputting a number is not difficult, so we may not have to generate numbers. But still, offering numbers as completions provides the information that a number is expected in that context. -->